agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/profiler/triggers/AlertTriggerSpanExporter.java

```diff
@@ -0,0 +1,151 @@
+import com.microsoft.applicationinsights.alerting.analysis.TimeSource;
+import com.microsoft.applicationinsights.alerting.analysis.data.TelemetryDataPoint;
+import com.microsoft.applicationinsights.alerting.config.AlertMetricType;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import java.util.Collection;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Intercepts spans, filters out those that have had alerts configured, and feeds them to an
+ * appropriate AlertPipeline.
+ */
+public class AlertTriggerSpanExporter implements SpanExporter {
+  private static final Logger LOGGER = LoggerFactory.getLogger(AlertTriggerSpanExporter.class);
+
+  // Next SpanExporter in App insights pipeline
+  private final SpanExporter delegate;
+
+  // Execution context of the execution loop
+  private final Future<?> spanProcessor;
+
+  // Spans to be processed
+  private final BlockingQueue<SpanData> queue;
+  private final Supplier<AlertingSubsystem> alertingSubsystemSupplier;
+
+  @Override
+  public CompletableResultCode export(Collection<SpanData> spans) {
+    // Process spans off of the main thread to avoid any complex logic blocking span processing
+    spans.forEach(queue::offer);
+
+    return delegate.export(spans);
+  }
+
+  @Override
+  public CompletableResultCode flush() {
+    int maxWaitCount = 10;
+    while (queue.size() > 0 && maxWaitCount-- > 0) {
+      try {
+        Thread.sleep(100);
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
+      }
+    }
+    return delegate.flush();
+  }
+
+  @Override
+  public CompletableResultCode shutdown() {
+    spanProcessor.cancel(false);
+    return delegate.shutdown();
+  }
+
+  /** Runnable that processes incoming spans off of the main thread. */
+  private class ProcessSpans implements Runnable {
+    @Override
+    public void run() {
+      while (true) {
+        try {
+          SpanData span = queue.poll(Long.MAX_VALUE, TimeUnit.SECONDS);
+          if (span != null) {
+            processSpan(span);
+          }
+        } catch (RuntimeException e) {
+          // Avoid killing polling due to a single exception
+          LOGGER.error("Error while processing span: {}", e.getMessage());
+        } catch (InterruptedException e) {
+          return;
+        }
+      }
+    }
+  }
+
+  public void processSpan(SpanData span) {
+    if (span.getKind() == SpanKind.SERVER) {
```

Comment:
I sent a commit to your branch to refactor `SpanDataMapper` and introduced a new method that you may want to use here

```suggestion
    if (SpanDataMapper.isRequest(span)) {
```